crates/bin/node/src/logic/inputs.rs

```diff
@@ -136,7 +151,20 @@ pub async fn process_swap_inputs<'a>(
 
         let keyset_info = keyset_cache.get_keyset_info(conn, proof.keyset_id).await?;
 
-        let keyset_unit = keyset_info.1;
+        let keyset_unit = keyset_info.unit();
+
+        // Validate amount doesn't exceed max_order
+        let max_order = keyset_info.max_order();
+        let max_value = (1u64 << max_order) - 1;
```

Comment:
if max order is 4 that make max_value 15.
We want max_value to be 8. 
Which is 1u64 << (max_order-1)